### PR TITLE
Automatically Deploy Strapi

### PR DESCRIPTION
**Pull request by TheGrayStone**

### CODE REVIEW

File: .github/workflows/build.staging.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          mkdir -p ~/.ssh/
+          echo "$SSH_KEY" > ./deploy.key
+          sudo chmod 600 ./deploy.key
+          echo "$SSH_KNOWN_HOSTS" > ~/.ssh/known_hosts
+        shell: bash
+        env:
+          SSH_KEY: ${{secrets.SSH_KEY}}
+          SSH_KNOWN_HOSTS: ${{secrets.SSH_KNOWN_HOSTS}}
+      - name: Install PM2 for deployment
+        run: yarn add pm2
```
Comment (Olliwehr) [inline .github/workflows/build.staging.yml:71]: We already have a `pm2` dependency in the package.json, maybe a `yarn install` would be more explicit or global add?

File: projects/bp-strapi/ecosystem.config.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+module.exports = {
+  apps: [
+    {
+      name: 'BP-Gallery Strapi',
```
Comment (Olliwehr) [inline projects/bp-strapi/ecosystem.config.js:4]: Tricky, but dont know what could be better here :D

File: projects/bp-strapi/ecosystem.config.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      ref: 'origin/main',
+      host: 'bp.bad-harzburg-stiftung.de',
+      path: '/home/github/BP2021RH1',
+      repo: 'https://github.com/hpi-swa-lab/BP2021RH1',
+      'post-deploy': 'cd projects/bp-strapi && yarn install && yarn build:prod && yarn pm2 reload ecosystem.config.js --env production && yarn pm2 save && git checkout yarn.lock'
+    },
+    staging: {
+      key: 'deploy.key',
+      user: 'github',
+      ref: 'origin/staging',
```
Comment (Olliwehr) [inline projects/bp-strapi/ecosystem.config.js:24]: We should definitively add a branch protection then

File: projects/bp-strapi/ecosystem.config.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      autorestart: true,
+      max_restarts: 50,
+    }
+  ],
+  deploy: {
+    production: {
+      key: 'deploy.key',
+      user: 'github',
+      ref: 'origin/main',
+      host: 'bp.bad-harzburg-stiftung.de',
```
Comment (Olliwehr) [inline projects/bp-strapi/ecosystem.config.js:16]: The host also is mentioned in a env file, right?

File: projects/bp-strapi/ecosystem.config.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      'post-deploy': 'cd projects/bp-strapi && yarn install && yarn build:prod && yarn pm2 reload ecosystem.config.js --env production && yarn pm2 save && git checkout yarn.lock'
+    },
+    staging: {
+      key: 'deploy.key',
+      user: 'github',
+      ref: 'origin/staging',
+      host: 'bp.bad-harzburg-stiftung.de',
+      path: '/home/github/BP2021RH1-staging',
+      repo: 'https://github.com/hpi-swa-lab/BP2021RH1',
+      'post-deploy': 'cd projects/bp-strapi && yarn install && yarn build && yarn pm2 reload ecosystem.config.js --env staging && yarn pm2 save && git checkout yarn.lock'
```
Comment (Olliwehr) [inline projects/bp-strapi/ecosystem.config.js:28]: `yarn install --frozen-lockfile` beats the checkout afterwards i guess? 🤔

File: scripts/restore_backup.sh [hunk 6]
```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+
+############ VARIABLE DEFINITIONS ############
+MEDIA_DIR=/home/dev/BP2021RH1/projects/bp-strapi/public/uploads
```
Comment (Olliwehr) [inline scripts/restore_backup.sh:5]: After the pm2 introduction this will lie in `/home/github/...`, no?

**Review by Olliwehr — APPROVED** [review]
That looks pretty neat! 🚀

File: scripts/restore_backup.sh (hunk 6 above)
Comment (TheGrayStone) [reply scripts/restore_backup.sh:5]:
```suggestion
MEDIA_DIR=/home/github/BP2021RH1/source/projects/bp-strapi/public/uploads
```

**Pull request merged**